### PR TITLE
hwmv2: boards: x86: doc: Adjust common docs to new locations

### PR DESCRIPTION
**Pull request by golowanow**
Adjust x86/common doc location to HWMv2 move.

### CODE REVIEW

**Review by fabiobaltieri — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

File: boards/intel/index.rst [hunk 1]
```diff
@@ -7,4 +7,4 @@ Intel Corporation
    :maxdepth: 1
    :glob:
 
-   **/*
+   [!common]*/**/*
```
Comment (gmarull) [inline boards/intel/index.rst:10]: does this really work? AFAIK glob `[]` is a range, so it won't pick any folder starting with `c`, `o`, `m` or `n` here.

Comment (golowanow) [reply boards/intel/index.rst:10]: well, at least currently it works this way
https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/x86/index.rst

Comment (golowanow) [reply boards/intel/index.rst:10]: and with this change
https://builds.zephyrproject.io/zephyr/pr/69075/docs/boards/intel/index.html

Comment (gmarull) [reply boards/intel/index.rst:10]: maybe it works by chance, but this is still wrong

**Review by jhedberg — APPROVED** [review]

**Review by gmarull — CHANGES_REQUESTED** [review]
glob needs fixing

File: boards/intel/index.rst (hunk 1 above)
Comment (golowanow) [reply boards/intel/index.rst:10]:
```
~/zephyrproject/zephyr/boards/intel$ ls -la [!common]*/**/*
-rw-r--r-- 1 rootroot rootroot 1892 Feb 16 10:52 intel_adl/doc/index.rst
-rw-r--r-- 1 rootroot rootroot 2416 Feb 16 10:53 intel_ehl/doc/index.rst
-rw-r--r-- 1 rootroot rootroot 2272 Feb 16 10:52 intel_rpl/doc/index.rst
~/zephyrproject/zephyr/boards/intel$ ls -la */**/*
-rwxr-xr-x 1 rootroot rootroot 1487 Feb 15 12:22 common/scripts/build_grub.sh
-rw-r--r-- 1 rootroot rootroot 1892 Feb 16 10:52 intel_adl/doc/index.rst
-rw-r--r-- 1 rootroot rootroot 2416 Feb 16 10:53 intel_ehl/doc/index.rst
-rw-r--r-- 1 rootroot rootroot 2272 Feb 16 10:52 intel_rpl/doc/index.rst
```

Comment (gmarull) [reply boards/intel/index.rst:10]:
```
$ cd boards/indel
$ mkdir -p cintel_test/doc
$ touch cintel_test/doc/index.rst
$ ls -la */**/*
-rw-r--r--. 1 gmarull gmarull    0 Feb 16 11:49 cintel_test/doc/index.rst
-rw-r--r--. 1 gmarull gmarull 1904 Feb 16 11:47 intel_adl/doc/index.rst
-rw-r--r--. 1 gmarull gmarull 2440 Feb 16 11:47 intel_ehl/doc/index.rst
-rw-r--r--. 1 gmarull gmarull 2284 Feb 16 11:47 intel_rpl/doc/index.rst
$ ls -la [!common]*/**/*
-rw-r--r--. 1 gmarull gmarull 1904 Feb 16 11:47 intel_adl/doc/index.rst
-rw-r--r--. 1 gmarull gmarull 2440 Feb 16 11:47 intel_ehl/doc/index.rst
-rw-r--r--. 1 gmarull gmarull 2284 Feb 16 11:47 intel_rpl/doc/index.rst
### OOOOPS cintel_test missing
```

**Comment by golowanow** [comment]
> glob needs fixing

please suggest how to fix it ?

**Comment by gmarull** [comment]
> > glob needs fixing
> 
> please suggest how to fix it ? (if it works here)

maybe use a folder starting with `_`, eg `_docs`? so then glob can be used, `[!_]*/...`

**Comment by golowanow** [comment]
> glob needs fixing

@finikorg referring your change, is there a better glob solution, what do you think ?
https://github.com/zephyrproject-rtos/zephyr/commit/0dab1e848233be8593d922070e1fd58ae8769df0#diff-8991089dacd6358c4805c86ffbb0d155176fd0550f9c36ce0422a59ef8f3a62dR10

**Comment by finikorg** [comment]
> > glob needs fixing
> 
> @finikorg referring your change, is there a better glob solution, what do you think ? [0dab1e8#diff-8991089dacd6358c4805c86ffbb0d155176fd0550f9c36ce0422a59ef8f3a62dR10](https://github.com/zephyrproject-rtos/zephyr/commit/0dab1e848233be8593d922070e1fd58ae8769df0#diff-8991089dacd6358c4805c86ffbb0d155176fd0550f9c36ce0422a59ef8f3a62dR10)

Probably rename `common` to `_common` and use `[!_]*/...`.

File: boards/intel/index.rst (hunk 1 above)
Comment (golowanow) [reply boards/intel/index.rst]:
```
~/zephyrproject/zephyr/boards/intel$ ls -la */**/index.rst
-rw-r--r-- 1 rootroot rootroot    0 Feb 16 12:16 cintel/doc/index.rst
-rw-r--r-- 1 rootroot rootroot 1892 Feb 16 10:52 intel_adl/doc/index.rst
-rw-r--r-- 1 rootroot rootroot 2416 Feb 16 10:53 intel_ehl/doc/index.rst
-rw-r--r-- 1 rootroot rootroot 2272 Feb 16 10:52 intel_rpl/doc/index.rst
```

**Review by gmarull — APPROVED** [review]
better with index usage, thanks

**Comment by finikorg** [comment]
Some boards might be missing in the documentation.

File: boards/up/up_squared_pro_7000/doc/up_squared_pro_7000.rst [hunk 2]
```diff
@@ -66,13 +66,13 @@ connection setup.
 Booting the UP Squared Pro 7000 Board using UEFI
 ================================================
```
Comment (finikorg) [inline boards/up/up_squared_pro_7000/doc/up_squared_pro_7000.rst:67]: You need to rename `{up_squared_pro_7000,index}.rst`

**Review by finikorg — CHANGES_REQUESTED** [review]

**Comment by golowanow** [comment]
> Some boards might be missing in the documentation.

which boards are missing ?
https://builds.zephyrproject.io/zephyr/pr/69075/docs/boards/up/index.html

File: boards/up/up_squared_pro_7000/doc/up_squared_pro_7000.rst (hunk 2 above)
Comment (golowanow) [reply boards/up/up_squared_pro_7000/doc/up_squared_pro_7000.rst:67]: > You need to rename `{up_squared_pro_7000,index}.rst`

why ? it is in different directory now `boards\up` and the pattern change is at `boards\intel\`

**Comment by finikorg** [comment]
> > Some boards might be missing in the documentation.
> 
> @finikorg which boards are missing ? https://builds.zephyrproject.io/zephyr/pr/69075/docs/boards/up/index.html

Ah, yes, they are now in the different directories, looks OK, although sourcing common docs from `intel` path does not look very nice.

**Review by finikorg — APPROVED** [review]

File: boards/up/up_squared_pro_7000/doc/up_squared_pro_7000.rst (hunk 2 above)
Comment (finikorg) [reply boards/up/up_squared_pro_7000/doc/up_squared_pro_7000.rst:67]: Yes, right, anyway would make sense to rename to make it consistent, all other docs are named as `index.rst`.

**Pull request merged**